.github/workflows/presubmit.yml

```diff
@@ -98,6 +98,36 @@ jobs:
             exit 1
           fi
 
+  check_clang_tidy:
+    name: Check clang-tidy
+    runs-on: ubuntu-latest
+    needs: check_files
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+      - name: Get docker images
+        run: |
+          if [[ ${{needs.check_files.outputs.run_job}} == false ]]; then
+            exit 0
+          fi
+          # https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio
+          echo $CR_PAT | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          docker pull ghcr.io/taichi-dev/taichidev-cpu-ubuntu18.04:v0.1.0
+        env:
+          CR_PAT: ${{ secrets.GITHUB_TOKEN }}
+      - name: Run clang-tidy
+        run: |
+          if [[ ${{needs.check_files.outputs.run_job}} == false ]]; then
+            exit 0
+          fi
+          docker run -id --user dev --name check_clang_tidy ghcr.io/taichi-dev/taichidev-cpu-ubuntu18.04:v0.1.0 /bin/bash
+          tar -cf - ../${{ github.event.repository.name }} --mode u=+rwx,g=+rwx,o=+rwx --owner dev --group dev | docker cp - check_clang_tidy:/home/dev/
+          docker exec --user root check_clang_tidy apt install -y clang-tidy-10
```

Comment:
Can we add a todo here saying that we should install clang-tidy-10 inside the docker image when we rebuild ;) ?

Reply:
Finished! Thanks :)